simpleui-server/src/sui_data.ts

```diff
@@ -112,235 +124,164 @@ export class SuiData {
-
-            // Callback invoked after zmqDataRequester.send() completes
-            zmqDataRequester.on('message', function (reply) {
-                const zmqResponse = SuiData.addXmlStatus(reply.toString());
-                zmqDataRequester.close();
-                SuiData.sendResponse(req, res, uiProps, zmqResponse);
-
-                Logger.log((SuiData.requestNum <= 5) ? LogLevel.INFO : LogLevel.DEBUG,
-                    `Request #${SuiData.requestNum} - zmqResponse: ` +
-                    `${zmqResponse.substr(0, 105).replace(/[ \t]*\n[ \t]*/g, '')}...`);
-                return;
-            });
+        // get the socket
+        const socket = SuiData.zmqMap.get(zmq_port);
 
-            // Callback invoked after zmqDataRequester gets an error
-            zmqDataRequester.on('error', function (zmqErr) {
-                const zmqResponse = ServerUtil.getServerError('ZMQ_ERROR', '{{ERROR}}', zmqErr);
-                SuiData.sendResponse(req, res, uiProps, zmqResponse);
-                Logger.log(LogLevel.WARNING,
-                    `Request #${SuiData.requestNum} - suiDataRequest(): ${zmqResponse}`);
-                zmqDataRequester.close();
-                return;
-            });
+        // get and set connection timeout (is this needed anymore? better to replace with reply timeout)
+        const timeout = SuiData.propOrDefault(uiProps, 'zmqTimeout', 1000);
+        socket.set_timeout(timeout);
 
-            // Synchronous processing that occurs PRIOR TO the callbacks above
-            zmqDataRequester.connect(`tcp://svcmachineapps:${SuiData.getZmqPort(req)}`);
-            const xmlDataRequestForZmq = `<request COMMAND="${cmd.cmd}" valueName="${cmd.valueName}"/>`;
-            Logger.log(LogLevel.DEBUG, `Send Request: ${xmlDataRequestForZmq}`);
-            Logger.log(LogLevel.DEBUG, `Port number of request: ${SuiData.getZmqPort(req)}`);
-            zmqDataRequester.send(xmlDataRequestForZmq);
-
-            // Callback invoked after SIGINT
-            process.on('SIGINT', function () {
-                const zmqResponse = ServerUtil.getServerError('ZMQ_ERROR', '{{ERROR}}', 'SIGINT');
-                SuiData.sendResponse(req, res, uiProps, zmqResponse);
-                Logger.log(LogLevel.WARNING,
-                    `Request #${SuiData.requestNum} - suiDataRequest() received SIGINT: ` + zmqResponse);
-                zmqDataRequester.close();
-            });
-        }
-        return;
-    }
+        // log zmq details
+        Logger.log( SuiData.requestNum <= 5 ? LogLevel.INFO : LogLevel.DEBUG, 
+            `zmq details:\n\ttimeout:\t${timeout}\n\tPort:\t\t${zmq_port}\n\tCMD:\t\t${JSON.stringify(cmd)}\n\tMsg:\t\t${zmq_xml_data_request_msg}`);
 
-    static createXmlResponse() {
-        return 'not yet implemented';
+        // send the data request
+        socket.send(zmq_xml_data_request_msg); 
+        
+        // add res and req to queue
+        socket.http_queue.enqueue([res, req]);
```

Comment:
...though actually, it looks like you're sending the ZMQ request from the HTTP handler here? Any reason you're doing that in this function?

Reply:
This is the way it was previously. The workflow makes sense to me. Simpleui-server only requests zmq data when base_app asks for it. Do you think theres a better way to do this?